app/models/edition.rb

```diff
@@ -244,21 +244,11 @@ def self.without_locked_documents
   end
 
   def self.latest_edition
-    where("NOT EXISTS (
-      SELECT 1
-        FROM editions e2
-       WHERE e2.document_id = editions.document_id
-         AND e2.id > editions.id
-         AND e2.state <> 'deleted')")
+    joins(:document).where("editions.id = documents.latest_edition_id")
   end
 
   def self.latest_published_edition
-    published.where("NOT EXISTS (
-      SELECT 1
-        FROM editions e2
-       WHERE e2.document_id = editions.document_id
-         AND e2.id > editions.id
-         AND e2.state = 'published')")
+    joins(:document).where("editions.id = documents.published_edition_id")
```

Comment:
The behaviour here has subtly changed. PReviously it would now have passed back an edition that is withdrawn, only published. However, i've been through the code and this class method isn't actually being used anywhere. I've left it in as it might be useful for scoping etc. when querying the db.

It could potentially be pulled out though.